bundles/binding/org.openhab.binding.temperlan/src/main/java/org/openhab/binding/temperlan/internal/TemperlanBinding.java

```diff
@@ -0,0 +1,228 @@
+		}
+	}
+
+	private Collection<TemperlanBindingConfig> getDeviceConfigs(
+			String deviceUid) {
+		Map<String, TemperlanBindingConfig> items = new HashMap<String, TemperlanBindingConfig>();
+		for (TemperlanBindingProvider provider : this.providers) {
+			provider.getDeviceConfigs(deviceUid, items);
+		}
+		return items.values();
+	}
+
+	private void sendUpdate(Collection<TemperlanBindingConfig> configs, BindingType type, State state) {
+		for (TemperlanBindingConfig config : configs) {
+			if (config.getBindingType() == type) {
+				eventPublisher.postUpdate(config.getItemName(), state);
+			}
+		}
+	}
+
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		
+		TemperlanBindingConfig config = getConfigForItemName(itemName);
+		if (config == null) {
+			logger.error("Received command for unknown item '" + itemName + "'");
+			return;
+		}
+		
+		TemperlanProxy proxy = proxies.get(config.getDeviceUid());
+		if (proxy == null) {
+			logger.error("Received command for unknown device uid '"
+					+ config.getDeviceUid() + "'");
+			return;
+		}
+
+		if (logger.isDebugEnabled()) {
+			logger.debug(BINDING_NAME + " processing command '" + command
+					+ "' of type '" + command.getClass().getSimpleName()
+					+ "' for item '" + itemName + "'");
+		}
+	}
+
+	private TemperlanBindingConfig getConfigForItemName(String itemName) {
+		for (TemperlanBindingProvider provider : this.providers) {
+			if (provider.getItemConfig(itemName) != null) {
+				return provider.getItemConfig(itemName);
+			}
+		}
+		return null;
+	}
+
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+		// ignore
+	}
+
+	@Override
+	public void updated(Dictionary<String, ?> config)
+			throws ConfigurationException {
+		logger.debug(BINDING_NAME + " updated");
+		try {
+			// Process device configuration
+			if (config != null) {
+				String refreshIntervalString = (String) config.get("refresh");
+				if (StringUtils.isNotBlank(refreshIntervalString)) {
+					refreshInterval = Long.parseLong(refreshIntervalString);
+				}
+				// parse all configured receivers
+				// ( temperlan:<uid>.host=10.0.0.2 )
+				// ( temperlan:<uid>.username=admin )
+				// ( temperlan:<uid>.password=pass )
+				
+				Enumeration<String> keys = config.keys();
+				while (keys.hasMoreElements()) {
+					String key = keys.nextElement();
+					if (key.endsWith(CONFIG_KEY_HOST)) {
+						// parse host
+						String host = (String) config.get(key);
+
+						int separatorIdx = key.indexOf('.');
+						// no uid => one device => use default UID
+						String uid = separatorIdx == -1 ? DEFAULT_DEVICE_UID : key.substring(0, separatorIdx);
+												
+						// proxy is stateless. keep them in a map in the
+						// binding.
+						proxies.put(uid, new TemperlanProxy(host));
+					}
+				}
+				setProperlyConfigured(true);
+			}
+		} catch (Throwable t) {
+			logger.error("Error configuring " + getName(), t);
+		}

```

Comment:
Consider putting the `try/catch` inside the big `if` block to improve readability.